caffeine@patapon.info/mprisMediaPlayer2.js

```diff
@@ -0,0 +1,342 @@
+    }
+
+    /**
+     * Set a callback function to isPlaying status changes.
+     * Use `disconnectIsPlaying(connectId)` to disconnect manually or `MprisPlayer.Destroy()`
+     * to destroy all connections.
+     * @param {(isPlaying: boolean) => void} callbackFn Callback function
+     * @returns {any}
+     */
+    connectIsPlaying(callbackFn) {
+        const connectId = this.connect(
+            'isPlaying',
+            (_, isPlaying) => callbackFn(isPlaying)
+        );
+        this._connections.add(connectId);
+        return connectId;
+    }
+
+    /**
+     * Manually disable a single `connectIsPlaying` connection.
+     * Run `MprisPlayer.Destroy()` to cleanup all connections.
+     * @param {any} connectId The `connectId` recieved from `connectIsPlaying()`.
+     * @returns {void}
+     */
+    disconnectIsPlaying(connectId) {
+        if (!this._connections.has(connectId)) {
+            return;
+        }
+        this.disconnect(connectId);
+        this._connections.delete(connectId);
+        return connectId;
+    }
+
+    _emitPlayStatus(forceEmit = false) {
+        if (this._lastEmittedPlayStatus === this.isPlaying && !forceEmit) {
+            return;
+        }
+        this._lastEmittedPlayStatus = this.isPlaying;
+        this.emit('isPlaying', this.isPlaying);
+    }
+
+    /**
+     * @param {string} dbusName Name in dbus
+     */
+    _addPlayer(dbusName) {
+        if (this._activePlayers.has(dbusName)) {
+            return;
+        }
+
+        const dbusPlayerProxy = new this._DBusPlayerProxy(
+            Gio.DBus.session,
+            dbusName,
+            '/org/mpris/MediaPlayer2',
+            (_player) => this._onPlayerChange()
+        );
+
+        const handlerId = dbusPlayerProxy.connect(
+            'g-properties-changed',
+            (_player) => this._onPlayerChange()
+        );
+
+        this._activePlayers.set(dbusName, {
+            handlerId,
+            playerProxy: dbusPlayerProxy
+        });
+    }
+
+    /**
+     * @param {string} dbusName Name in dbus
+     */
+    _removePlayer(dbusName) {
+        const player = this._activePlayers.get(dbusName);
+        if (!player) {
+            return;
+        }
+        player.playerProxy.disconnect(player.handlerId);
+        this._activePlayers.delete(dbusName);
+    }
+
+    _onPlayerChange() {
+        let isPlaying = false;
+        for (const player of this._activePlayers.values()) {
+            if (player.playerProxy.PlaybackStatus === 'Playing') {
+                isPlaying = true;
+            }
+        }
+        this._isPlaying = isPlaying;
+        this._emitPlayStatus();
+    }
+
+    _onNameOwnerChanged(_proxy, _sender, [name, oldOwner, newOwner]) {
+        if (!name.startsWith(this._mprisPrefix)) {
+            return;
+        }
+        if (oldOwner && !newOwner) {
+            this._removePlayer(name);
+        }
+        if (newOwner && !oldOwner) {
+            this._addPlayer(name);
+        }
```

Comment:
Is something like this a bit clearer?

```suggestion
        if (oldOwner !== newOwner) {
            if (newOwner) {
                this._addPlayer(name);
            } else if (oldOwner) {
                this._removePlayer(name);
            }
        }
```

If they're guaranteed that both are never set at the same time, `if (oldOwner !== newOwner)` could also be dropped in that

Reply:
I don't know. Tbh I could not find clear documentation for this. All I know is that code, as is, works and is tested.

 `oldOwner !== newOwner` can these be compared? 
`If they're guaranteed that both are never set at the same time` don't know if that' s guaranteed.

All I found was that if there is an `oldOwner` and no `newOwner` that means an app is removed and visa versa. So I implemented and tested this.

Reply:
Found this https://dbus.freedesktop.org/doc/dbus-specification.html#bus-messages-name-owner-changed

Seems that if `newOwner` isn't set then it was removed and if `oldOwner` isn't set it was added

```suggestion
    if (newOwner === "") {
        this._removePlayer(name);
    } else if (oldOwner === "") {
        this._addPlayer(name);
    }
```

Completely untested, I don't know if they get set to `undefined`, `null` or `""` but in principle, that should work

Reply:
I tested your suggestion following the dbus spec and commited the change.